components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -1059,7 +1074,55 @@ public static boolean revokeTokens(String username, UserStoreManager userStoreMa
         }
 
         boolean isErrorOnRevokingTokens;
+
+        // This is for FEDERATED users from sub organization.
+        String userResidentTenant = null;
+        String authenticatedUserName = null;
+        if (authenticatedUser.getUserResidentOrganization() != null) {
+            try {
+                userResidentTenant = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                        .resolveTenantDomain(authenticatedUser.getUserResidentOrganization());
+                if (OrganizationManagementUtil.isOrganization(userResidentTenant)) {
+                    /*
+                     Keeping the username to be used to extract the tokens which are created directly from the
+                     sub organization applications.
+                    */
+                    authenticatedUserName = authenticatedUser.getUserName();
+                    authenticatedUser.setUserName(authenticatedUser.getUserId());
+                }
+            } catch (OrganizationManagementException | UserIdNotFoundException e) {
+                throw new UserStoreException("Error occurred while constructing the authenticated user.", e);
+            }
+        }
         isErrorOnRevokingTokens = processTokenRevocation(clientIds, authenticatedUser, userStoreDomain, username);
+        if (authenticatedUser.getUserResidentOrganization() != null) {
```

Comment:
Why `authenticatedUser.getUserResidentOrganization() != null`  logic is separately handled in 1098 and 1081 lines?
Can't we handled these in a single if block?